From 300535edd8848c5ff892959c0e4b534c7fc28458 Mon Sep 17 00:00:00 2001
From: German Cvetkov <german.cvetkov@lge.com>
Date: Fri, 18 Oct 2024 12:28:50 +0300
Subject: [PATCH] [build][webos] 131.0.6740.0~1: Disable AOM SVE for webOS

:Release Notes:
Disabled AOM SVE for webOS to fix build error
after rolling to 131.0.6740.0~1

:Detailed Notes:
In upstream CL http://crrev.com/c/5888468 enables SVE and SVE2
to use in AOM but webOS toolchain does not have it.

Without this change the build fails with compile error:
| ../../../../../../../../../mnt/sda/neva-rolling/chromium/src/third_party/libaom/source/libaom/aom_dsp/arm/aom_neon_sve_bridge.h:15:10: fatal error: arm_neon_sve_bridge.h: No such file or directory
|    15 | #include <arm_neon_sve_bridge.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~~~

:Issues Addressed:
[NEVA-9673] Upgrade neva-main-rolling up to 6740 BUILD
---
 aom_ports/arm.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/aom_ports/arm.h b/aom_ports/arm.h
index d1d47692d5..c3ebfe99b1 100644
--- a/aom_ports/arm.h
+++ b/aom_ports/arm.h
@@ -27,10 +27,15 @@ extern "C" {
 #define HAS_NEON_DOTPROD (1 << 2)
 // Armv8.2-A optional Neon i8mm instructions, mandatory from Armv8.6-A.
 #define HAS_NEON_I8MM (1 << 3)
+#if !defined(OS_WEBOS)
 // Armv8.2-A optional SVE instructions, mandatory from Armv9.0-A.
 #define HAS_SVE (1 << 4)
 // Armv9.0-A SVE2 instructions.
 #define HAS_SVE2 (1 << 5)
+#else
+#define HAS_SVE 0
+#define HAS_SVE2 0
+#endif
 
 int aom_arm_cpu_caps(void);
 
-- 
2.34.1

